pvlib/bifacial/infinite_sheds.py

```diff
@@ -0,0 +1,831 @@
+        sky from the ground. [W/m^2]
+    f_gnd_pv_shade : numeric
+        fraction of ground visible from shaded part of PV surface. [unitless]
+    f_gnd_pv_noshade : numeric
+        fraction of ground visible from unshaded part of PV surface. [unitless]
+
+    Returns
+    -------
+    numeric
+        Ground diffuse irradiance on the row plane. [W/m^2]
+    """
+    return poa_ground * (f_x * f_gnd_pv_shade + (1 - f_x) * f_gnd_pv_noshade)
+
+
+def _shaded_fraction(apparent_zenith, azimuth, surface_tilt,
+                     surface_azimuth, gcr):
+    """
+    Calculate fraction (from the bottom) of row slant height that is shaded
+    from direct irradiance by the row in front toward the sun.
+
+    See [1], Eq. 14 and also [2], Eq. 32.
+
+    .. math::
+        F_x = \\max \\left( 0, \\min \\left(\\frac{\\text{GCR} \\cos \\theta
+        + \\left( \\text{GCR} \\sin \\theta - \\tan \\beta_{c} \\right)
+        \\tan Z - 1}
+        {\\text{GCR} \\left( \\cos \\theta + \\sin \\theta \\tan Z \\right)},
+        1 \\right) \\right)
+
+    Parameters
+    ----------
+    apparent_zenith : numeric
+        Apparent solar zenith. [degrees]
+    azimuth : numeric
+        Solar azimuth. [degrees]
+    surface_tilt : numeric
+        Row tilt from horizontal, e.g. surface facing up = 0, surface facing
+        horizon = 90. [degrees]
+    surface_azimuth : numeric
+        Azimuth angle of the row surface. North=0, East=90, South=180,
+        West=270. [degrees]
+    gcr : numeric
+        Ground coverage ratio, which is the ratio of row slant length to row
+        spacing (pitch). [unitless]
+
+    Returns
+    -------
+    f_x : numeric
+        Fraction of row slant height from the bottom that is shaded from
+        direct irradiance.
+
+    References
+    ----------
+    .. [1] Mikofski, M., Darawali, R., Hamer, M., Neubert, A., and Newmiller,
+       J. "Bifacial Performance Modeling in Large Arrays". 2019 IEEE 46th
+       Photovoltaic Specialists Conference (PVSC), 2019, pp. 1282-1287.
+       :doi:`10.1109/PVSC40753.2019.8980572`.
+    .. [2] Kevin Anderson and Mark Mikofski, "Slope-Aware Backtracking for
+       Single-Axis Trackers", Technical Report NREL/TP-5K00-76626, July 2020.
+       https://www.nrel.gov/docs/fy20osti/76626.pdf
+    """
+    tan_phi = utils._solar_projection_tangent(
+        apparent_zenith, azimuth, surface_azimuth)
+    # length of shadow behind a row as a fraction of pitch
+    x = gcr * (sind(surface_tilt) * tan_phi + cosd(surface_tilt))
+    f_x = 1 - 1. / x
+    # set f_x to be 1 when sun is behind the array
+    ao = aoi(surface_tilt, surface_azimuth, apparent_zenith, azimuth)
+    f_x = np.where(ao < 90, f_x, 1.)
+    # when x < 1, the shadow is not long enough to fall on the row surface
+    f_x = np.where(x > 1., f_x, 0.)
+    return f_x
+
+
+def get_irradiance_poa(surface_tilt, surface_azimuth, apparent_zenith,
+                       azimuth, gcr, height, pitch, ghi, dhi, dni,
```

Comment:
Given where #1403 is going, I suggest we go with `solar_zenith` and `solar_azimuth` like we do here

https://github.com/pvlib/pvlib-python/blob/c0c46b4dcb5b34f4062c8160d2c92d4c5089b273/pvlib/irradiance.py#L304-L305

Reply:
Should we wait until after #1403 is closed to merge this PR?